libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/usecase/route/FetchRouteUseCase.kt

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.navigation.dropin.usecase.route
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.extensions.coroutines.requestRoutes
+import com.mapbox.navigation.dropin.usecase.UseCase
+import com.mapbox.navigation.dropin.usecase.location.GetCurrentLocationUseCase
+import com.mapbox.navigation.utils.internal.toPoint
+import kotlinx.coroutines.CoroutineDispatcher
+
+internal class FetchRouteUseCase(
+    private val navigation: MapboxNavigation,
+    private val routeOptionsBuilder: () -> RouteOptions.Builder,
+    private val getCurrentLocation: GetCurrentLocationUseCase,
+    dispatcher: CoroutineDispatcher
+) : UseCase<Point, List<DirectionsRoute>>(dispatcher) {
+
+    override suspend fun execute(destination: Point): List<DirectionsRoute> {
+        val origin = getCurrentLocation(Unit).getOrNull() ?: return emptyList()
+
+        val routeOptions = routeOptionsBuilder()
+            .coordinatesList(listOf(origin.toPoint(), destination))
+            .layersList(listOf(navigation.getZLevel(), null))
+            .alternatives(true)
+            .build()
+
+        return navigation.requestRoutes(routeOptions).routes
```

Comment:
This doesn't handle the exception that could be thrown in case of a request failure, refs https://github.com/mapbox/mapbox-navigation-android/pull/5506#discussion_r818536387.

Reply:
Superclass handles it.  [UseCase#invoke](https://github.com/mapbox/mapbox-navigation-android/blob/5b25af64d74897fe67e2886828dd5fe26f10414e/libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/usecase/UseCase.kt#L46)

Reply:
Since we are not going with UseCase approach, I am removing this class. This is no longer valid.